platform_specific/ubuntu/YamlParameters.cpp

```diff
@@ -22,40 +35,82 @@ void YamlParameters::read_from_file(uint8_t* flash_memory,
-            memcpy(flash_memory + 4 * int_param_idx, &int_param_value, 4);
+            std::string num_value;
+            std::remove_copy_if(value.begin(), value.end(), num_value.begin(), ::isspace);
+            int32_t int_value = std::stoi(num_value);
+            memcpy(flash_memory + 4 * int_param_idx, &int_value, 4);
             int_param_idx++;
         } catch (std::invalid_argument const& ex) {
-            size_t offset = 2048 - MAX_STRING_LENGTH * (1 + str_param_idx);
-            memcpy(flash_memory + offset, value.c_str(), sizeof(value));
+            size_t quote_pos = value.find('"');
+            size_t quote_end_pos = value.find('"', quote_pos + 1);
+            std::string str_value = value.substr(quote_pos + 1, quote_end_pos - quote_pos - 1);
+            int offset = pages_n * 2048 - MAX_STRING_LENGTH *
+                                                        (NUM_OF_STR_PARAMS - str_param_idx);
+            memcpy(flash_memory + offset, str_value.c_str(), strlen(str_value.c_str()));
+            memcpy(flash_memory + offset + strlen(str_value.c_str()), "\0", 1);
             str_param_idx++;
         }
     }
+    std::get<0>(*last_idxs) = int_param_idx;
+    std::get<1>(*last_idxs) = str_param_idx;
+    params_storage_file.close();
+    return LIBPARAMS_OK;
 }
 
-void YamlParameters::write_to_file(uint8_t* flash_memory,
-                                   std::ofstream& params_storage_file) {
-    for (size_t index = 0; index < IntParamsIndexes::INTEGER_PARAMS_AMOUNT; index++) {
+int8_t YamlParameters::write_to_file(std::string path, uint8_t* flash_memory, size_t pages_n,
+                                        std::tuple<uint8_t, uint8_t>* last_idxs, size_t page_size) {
+    if (flash_memory == NULL || pages_n < 1 || path.empty() || page_size < 4) {
+        return LIBPARAMS_WRONG_ARGS;
+    }
+
+    std::ofstream params_storage_file;
+    params_storage_file.open(path, std::ios_base::out);
+
+    std::cout << "YamlParameters: save data to " << path << std::endl;
+    uint16_t n_bytes = 0;
+    for (uint8_t index = std::get<0>(*last_idxs); index < IntParamsIndexes::INTEGER_PARAMS_AMOUNT;
+                                                                                        index++) {
         int32_t int_param_value;
         memcpy(&int_param_value, flash_memory + index * 4, 4);
         params_storage_file << std::left << std::setw(32) << integer_desc_pool[index].name << ":\t"
                             << int_param_value << "\n";
         std::cout << std::left << std::setw(32) << integer_desc_pool[index].name
-                  << ":\t" << int_param_value << "\n";
+                                                            << ":\t" << int_param_value << "\n";
+        n_bytes += 4;
+        std::get<0>(*last_idxs) = index + 1;
+        if (n_bytes > 2048 - 4) {
+            break;
+        }
+    }
+
+    auto last_str_idx = std::get<1>(*last_idxs);
+    auto num_str_params = NUM_OF_STR_PARAMS;
```

Comment:
Okay, but should we add the integer_desc_pool and string_desc_pool to YamlParameters constructor? 
Because 
1-we can make tests more flexible, 
2- NUM_OF_STR_PARAMS and INTEGER_PARAMS_AMOUNT depends on those arrays